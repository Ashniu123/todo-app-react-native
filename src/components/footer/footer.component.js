import React, { Component } from 'react';
import { View, Button, Alert } from 'react-native';
import { connect } from 'react-redux';

import { startRemoveCompleted } from '../../actions/tasks';
import styles from './footer.style';

export class Footer extends Component {
  constructor(props) {
    super(props);

    this.handleOnPress = this.handleOnPress.bind(this);
  }

  handleOnPress() {
    this.props.startRemoveCompleted();
  }

  render() {
    return (
      <View style={styles.footer__view}>
        <Button
          color="#ef5350"
          title="Remove completed items"
          accessibilityLabel="Remove completed items"
          onPress={this.handleOnPress}
        />
      </View>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startRemoveCompleted: () => dispatch(startRemoveCompleted())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Footer);
