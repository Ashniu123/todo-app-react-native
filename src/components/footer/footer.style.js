import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  footer__view: {}
});
