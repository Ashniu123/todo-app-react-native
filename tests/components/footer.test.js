import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from '../../src/components/footer/footer.component';

let wrapper, startRemoveCompleted;

beforeEach(() => {
  startRemoveCompleted = jest.fn();
  wrapper = shallow(<Footer startRemoveCompleted={startRemoveCompleted} />);
});

test('should render Footer component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should remove all completed items on button press', () => {
  wrapper.find('Button').prop('onPress')();
  expect(startRemoveCompleted).toHaveBeenCalled();
});
