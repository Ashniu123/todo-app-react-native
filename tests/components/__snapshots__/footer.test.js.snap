// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer component correctly 1`] = `
<Component
  style={Object {}}
>
  <Button
    accessibilityLabel="Remove completed items"
    color="#ef5350"
    onPress={[Function]}
    title="Remove completed items"
  />
</Component>
`;
